webapi/src/main/scala/org/knora/webapi/messages/v2/routing/authenticationmessages/AuthenticationMessagesV2.scala

```diff
@@ -33,11 +33,25 @@ import spray.json._
   * @param identifier   the user's IRI, username, or email.
   * @param password     the user's password.
   */
-case class LoginApiRequestPayloadV2(identifier: String,
-                                    password: String) {
+case class LoginApiRequestPayloadV2(iri: Option[IRI] = None,
```

Comment:
All these inputs (IRI, email, and username) are inserted into SPARQL queries, right? If so, they should be escaped using `StringFormatter.toSparqlEncodedString()` to prevent SPARQL injection attacks.

Reply:
ok, so the general workflow would be:

1. validate
2. StringFormatter.toSparqlEncodedString()

Correct?

Reply:
* For IRIs: `StringFormatter.validateAndEscapeIRI()`
* For email addresses: `StringFormatter.validateEmail()`, then `StringFormatter.toSparqlEncodedString()` (you could make a `validateAndEscapeEmail` function to do both)
* For usernames: Are there any restrictions on what characters can be in a username? Maybe there should be. Whatever restrictions you choose, validate, then `StringFormatter.toSparqlEncodedString()` (you could make a `validateAndEscapeUsername` function to do both)

Reply:
Also maybe `StringFormatter` should have a maximum allowed length for IRIs, email addresses, and usernames.